lib/ansible/modules/cloud/amazon/ecs_ecr.py

```diff
@@ -350,31 +344,26 @@ def run(ecr, params, verbosity):
 
 
 def main():
-    argument_spec = ec2_argument_spec()
-    argument_spec.update(dict(
+    argument_spec = dict(
         name=dict(required=True),
         registry_id=dict(required=False),
         state=dict(required=False, choices=['present', 'absent'],
                    default='present'),
         force_set_policy=dict(required=False, type='bool', default=False),
         policy=dict(required=False, type='json'),
-        delete_policy=dict(required=False, type='bool')))
-
-    module = AnsibleModule(argument_spec=argument_spec,
-                           supports_check_mode=True,
-                           mutually_exclusive=[
-                               ['policy', 'delete_policy']])
+        delete_policy=dict(required=False, type='bool'))
 
-    if not HAS_BOTO3:
-        module.fail_json(msg='boto3 required for this module')
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[['policy', 'delete_policy']])
 
-    ecr = EcsEcr(module)
-    passed, result = run(ecr, module.params, module._verbosity)
+    passed, result = run(EcsEcr(module), module.params, module._module._verbosity)
 
     if passed:
-        module.exit_json(**result)
+        module.exit_json(**camel_dict_to_snake_dict(result))
```

Comment:
@s-hertel / @jillr : Technically this is a breaking change, but it brings things in line with the Python/Ansible norms and the module is flagged as 'preview' thoughts?

Reply:
Indeed, it's flagged preview :)